Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -579,7 +606,84 @@ private extension HTTPClient {
             }
         }
     }
+}
+
+// MARK: - Request Retry Logic
+extension HTTPClient {
```

Comment:
This extension has some logic that might be better suited to its own file, such that the only stuff that's remaining in HTTPClient is `retryRequestIfNeeded`

Reply:
Doesn't need to be addressed in this PR, but would be a good follow-up